stl/src/xfdscale.cpp

```diff
@@ -37,7 +37,7 @@ short _FDscale(float* px, long lexp) { // scale *px by 2^xexp with checking
             unsigned short psx = 0;
 
             if (xexp <= -16) { // scale by words
-                psx          = ps->_Sh[_F1] | (psx != 0 ? 1 : 0);
+                psx          = ps->_Sh[_F1] | 0;

```

Comment:
I agree that this change is correct (thanks!) but it should be further simplified. Anything bitwise-ORed with `0` is unchanged, so the `| 0` should be dropped completely.

Aside: It appears that this was a relic of the original code for `float` being based on the code for `double` where there's a `for` loop instead of an `if` statement:
https://github.com/microsoft/STL/blob/285187b7b24be4bebfa271fbd0b1eacd995d18f4/stl/src/xdscale.cpp#L41-L44